flocker/acceptance/test_api.py

```diff
@@ -305,6 +305,69 @@ def in_current():
         creating.addCallback(created)
         return creating
 
+    def assert_busybox_http(self, host, port):
+        """
+        Assert that a HTTP serving a response with body ``b"hi"`` is running
+        at given host and port.
+
+        :param host: Host to connect to.
+        :param port: Port to connect to.
+        """
+        def query(host, port):
+            req = get(
+                "http://{host}:{port}".format(host=host, port=port),
+                persistent=False
+            ).addCallback(content)
+            return req
+
+        d = verify_socket(host, port)
+        d.addCallback(lambda _: query(host, port))
+        d.addCallback(self.assertEqual, b"hi")
+        return d
+
+    @require_cluster(1)
+    def test_non_root_container_can_access_dataset(self, cluster):
+        """
+        A container running as a user that is not root can write to a
+        dataset attached as a volume.
+        """
+        _, port = find_free_port()
+        node = cluster.nodes[0]
+        container = {
+            u"name": random_name(self),
+            u"node_uuid": node.uuid,
+            u"image": u"busybox",
+            u"ports": [{u"internal": 8080, u"external": port}],
+            u'restart_policy': {u'name': u'never'},
+            u"volumes": [{u"dataset_id": None,
+                          u"mountpoint": u"/data"}],
+            u"command_line": [
+                # Run as non-root user:
+                u"su", u"-", u"nobody", u"-c", u"sh", u"-c",
+                # Write something to volume we attached, and then
+                # expose what we wrote as a web server:
+                u"""\
+echo -n '#!/bin/sh
+echo -n "HTTP/1.1 200 OK\r\n\r\nhi"
+' > /data/script.sh;
+chmod +x /data/script.sh;
+nc -ll -p 8080 -e /data/script.sh
```

Comment:
For some reason that's less info than man page inside the docker container. I added comment with reference.